torch/_decomp/decompositions.py

```diff
@@ -1371,6 +1371,29 @@ def addmv(self: Tensor, mat1: Tensor, vec: Tensor, beta: int = 1, alpha: int = 1
     return out + beta * self
 
 
+@register_decomposition(aten.linalg_cross)
+@out_wrapper()
+@pw_cast_for_opmath
+def linalg_cross(a: Tensor, b: Tensor, dim: int = -1):
+    a_, b_ = torch.broadcast_tensors(a, b)
+    dim = utils.canonicalize_dim(a_.ndim, dim)
+    a_0, a_1, a_2 = torch.split(a_, 1, dim=dim)
+    b_0, b_1, b_2 = torch.split(b_, 1, dim=dim)
+    return torch.cat(
+        [a_1 * b_2 - a_2 * b_1, a_2 * b_0 - a_0 * b_2, a_0 * b_1 - a_1 * b_0], dim=dim
+    )
+    # cross_idx = torch.arange(3, device=a_.device)
+    # return torch.cat([
+    #         a_.index_select(dim, cross_idx[(i + 1) % 3])
+    #         * b_.index_select(dim, cross_idx[(i + 2) % 3])
+    #         - a_.index_select(dim, cross_idx[(i + 2) % 3])
+    #         * b_.index_select(dim, cross_idx[(i + 1) % 3])
+    #         for i in range(3)
```

Comment:
Could we not avoid the cat if you wrote this as `a_.index_select(dim, (cross_idx + 1) % 3)` etc.

Reply:
nb. @amjames cat prevents fusion in many cases, so it'd be best to avoid it if we can.

On a separate note, I wonder whether we identify these sort of patterns and we convert them to ModularIndexing... We probably don't, so I don't know if we know how to generate efficient code for that pattern as-is.

Reply:
We don't atm but it could easily be added to `IndexPropagation`

Reply:
In this case `cat` seems to be the only way I can get this to generate a single fused kernel. I have tried a few different ways of creating the result tensor and assigning to slices of it all of them produce multiple kernels. 

The commented out version that I left in here actually generates identical code as the final version I left in. I replaced it because I thought it was a bit easier to read with all other things apparently being equal.

Reply:
@amjames what about the version I suggested above, we don't need to assign any slices.

Reply:
Note that `(cross_idx + 1) % 3` is a tensor so the indexing op will have the same shape as the output.